static-exporter/main.libsonnet

```diff
@@ -47,14 +47,17 @@ local k = import 'ksonnet-util/kausal.libsonnet';
     }),
 
   metric:: {
-    new(name, description)::
+    new(name, description, metricType='counter')::
```

Comment:
You could set this value below and require `withMetricType` instead of overloading the constructor.

Reply:
Is there a reason to prefer that over having all the arguments in the constructor?
I'm not sure I see how `metricType` is different from `description` or `name` except that the library originally only supported `counter` types.

In other words, why have:

```jsonnet
metric.new('NAME', 'DESCRIPTION') + metric.withMetricType('TYPE')
```

when you can just have:

```jsonnet
metric.new('NAME', 'DESCRIPTION, 'TYPE')
```

Reply:
Also, to help me understand our patterns better. Why does the constructor do:

```jsonnet
self.withName(name)
+ self.withDescription(description)
+ self.withMetricType(metricType)
```

instead of:

```jsonnet
{
  name: name,
  description: description,
  type: metricType,
}
```

I am thinking it's to use the public API presented by the object so that if `withName` ever does validation, the constructor also does the validation without refactor?
In that case, should we have some convention of indicating that the `type`, `description`, and `type` fields shouldn't be modified except through that public API. Perhaps they should be fields within a hidden `_internal` field or something?